core/src/tile/tileBuilder.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "data/dataSource.h"
+#include "scene/styleContext.h"
+#include "scene/drawRule.h"
+
+namespace Tangram {
+
+class DataLayer;
+class DataSource;
+class Tile;
+struct TileData;
+class StyleBuilder;
+
+class TileBuilder {
+
+public:
+
+    TileBuilder();
+
+    ~TileBuilder();
+
+    StyleBuilder* getStyleBuilder(const std::string& _name);
+
+    void setScene(std::shared_ptr<Scene> _scene);
+
+    std::shared_ptr<Tile> build(TileID _tileID, const TileData& _data, const DataSource& _source);
+
+    const Scene& scene() const { return *m_scene; }
+
+    Tile& tile() const { return *m_tile; }
+
+private:
+    std::shared_ptr<Scene> m_scene;
+
+    StyleContext m_styleContext;
+    DrawRuleMergeSet m_ruleSet;
+
+    std::vector<const DataLayer*> m_activeLayers;
```

Comment:
yeah - I forgot to add tileBuilder.h for a while in the large tile+style-builder branch... This is belongs tile-data-sink, the other to style-builder